src/octarr.cpp

```diff
@@ -78,6 +82,34 @@ OctetArray& OctetArray::operator=(const OctetArray& other) {
     return *this;
 }
 
+OctetArray& OctetArray::operator<<(const BitArray& bitarr) {
+    if (bitarr.length() < 1 || !bitarr.get_octets()) {
+        return *this;
+    }
+
+    // Reload contents of this OctetArray with the encoded octets from bit array
+    if (this->octets != nullptr) {
```

Comment:
This check could be removed since `!bitarr.get_octets()` is already a checking condition for `nullptr` indicating a failed allocation of the referenced `BitArray`.